tidy3d/components/parameter_perturbation.py

```diff
@@ -942,6 +942,12 @@ def sample(
         if isinstance(h_vals, UnstructuredGridDataset):
             h_vals = h_vals.values
 
+        # Needed to avoid error in some xarray / numpy versions
+        if e_vals.size == 1:
```

Comment:
just a heads up, not sure of the shape of `e_vals`, but if I do this float casting I get a deprecation warning (both in numpy 1 and 2)

```py
>>> import numpy as np
>>> a = np.array([[[[1]]]])
>>> float(a)
<python-input-3>:1: DeprecationWarning: Conversion of an array with ndim > 0 to a scalar is deprecated, and will error in future. Ensure you extract a single element from your array before performing this operation. (Deprecated NumPy 1.25.)
  float(a)
1.0

```

perhaps it would be safer to `float(np.squeeze(e_vals))` or I think `e_vals.item()` is perhaps the standard way.

Reply:
yea good point